From: SeongJae Park <sj@kernel.org>
Date: Wed, 5 Mar 2025 20:19:08 -0800
Subject: [PATCH] ==== memory tiering ====

mm/damon: add DAMOS quota goal metrics for auto-tuned memory tiering

DAMON is being used for memory tiering.  The currently available
solution does static access pattern-based control.  It requires tuning
on different workloads and systems.  Add DAMOS quota goal metrics for
making it auto-tuned for upper tier's headroom space.  Evaluation shows
12.6 percent improvement over NUMA balancing based tiering on a setup.

Backgrounds
===========

Memory Tiering Idea
-------------------

NUMA Balancing-based Memory Tiering
-----------------------------------

Pros: easy to use
Cons: page faults overhead, synchrnous migration overhead, statically
set thresholds

DAMON-based Manually-Tuned Memory Tiering
-----------------------------------------

Pros: minimum overhead, asynchronous migration
Cons: statically set thresholds

DAMOS Quota Goal Metrics for Auot-tuned Memory Tiering
======================================================

Evaluation
==========

Future Plans
============

Background

Patch series "DAMON based tiered memory management for CXL memory"
starting with commit a00ce85af2a1 ("mm: make alloc_demote_folio
externally invokable for migration") added migrate_{hot,cold} DAMOS
actions.

Patch sereis "mm/damon: let DAMOS feeds and tame/auto-tune itself"
starting with commit 78f2f60377ee ("mm/damon/core: set damos_quota->esz
as public field and document") allowed self-tuning DAMOS.

RFC idea for DAMOS-based tiered memory management:
https://lore.kernel.org/all/20231112195602.61525-1-sj@kernel.org/

Evaluation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_gpnk7cul | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_gpnk7cul

diff --git a/hkml_cv_bogus/hkml_cv_bogus_gpnk7cul b/hkml_cv_bogus/hkml_cv_bogus_gpnk7cul
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

